userspace/otpilot/src/main.rs

```diff
@@ -82,21 +84,13 @@ fn run() -> TockResult<()> {
         device_id: [0; 64],
     };
 
-    {
-        let mut idx : usize = 0;
-        for val in "v1.00".as_bytes() {
-            if idx > identity.version.len() { break; }
-            identity.version[idx] = *val;
-            idx = idx + 1;
-        }
-    }
+    let banner = "v1.00".as_bytes();
+    let max_len = min(identity.version.len(), banner.len());
+    identity.version[..max_len].copy_from_slice(&banner[..max_len]);
 
-    {
-        let dev_id_bytes = fuse::get().get_dev_id()?.to_be_bytes();
-        for idx in 0..dev_id_bytes.len() {
-            identity.device_id[idx] = dev_id_bytes[idx];
-        }
-    }
+    let dev_id_bytes = fuse::get().get_dev_id()?.to_be_bytes();
+    let max_len = min(identity.device_id.len(), dev_id_bytes.len());
```

Comment:
If `device_id_bytes` is too long, this will silently drop the tail end.  Is that OK?

Reply:
On Haven, dev_id_bytes will always be 8 bytes long, so not risk here. But apart from that, I don't think there's much we could do. We could print out an error message and continue ...

Reply:
If no risk, then it should be an assertion.  If `min()` is necessary, then at the very least we should print something if we truncate.

Silent failure is *evil*.

Reply:
Fair. I added code to print an error.